source/s3_default_meta_request.c

```diff
@@ -255,6 +255,7 @@ static void s_s3_meta_request_default_request_finished(
     AWS_PRECONDITION(meta_request_default);
 
     if (error_code == AWS_ERROR_SUCCESS && meta_request->headers_callback != NULL &&
+        request->send_data.response_headers != NULL &&
```

Comment:
trivial: remove this change, when the error code is success, there should always be response headers.

Reply:
Letting the check just to be 100% safe a manufactured response with status code 200 without headers wouldn't cause trouble.